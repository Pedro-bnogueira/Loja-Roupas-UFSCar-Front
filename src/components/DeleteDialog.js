import React from "react";
import {
    Dialog,
    DialogTitle,
    DialogContent,
    DialogContentText,
    DialogActions,
    Button,
} from "@mui/material";

const DeleteDialog = ({ open, onClose, onConfirm, title, description }) => {
    return (
        <Dialog open={open} onClose={onClose}>
            <DialogTitle>{title}</DialogTitle>
            <DialogContent>
                <DialogContentText>{description}</DialogContentText>
            </DialogContent>
            <DialogActions>
                <Button onClick={onClose} color="primary">
                    NÃ£o
                </Button>
                <Button onClick={onConfirm} color="error" autoFocus>
                    Sim
                </Button>
            </DialogActions>
        </Dialog>
    );
};

export default DeleteDialog;
