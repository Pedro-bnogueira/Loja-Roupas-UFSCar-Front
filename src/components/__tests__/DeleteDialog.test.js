import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import DeleteDialog from "../DeleteDialog";
import '@testing-library/jest-dom';

describe("DeleteDialog Component", () => {
  const mockOnClose = jest.fn();
  const mockOnConfirm = jest.fn();
  const title = "Confirmar Deleção";
  const description = "Tem certeza que deseja excluir este item?";

  const renderComponent = (open = true) => {
    return render( // <-- Retornando o resultado do render()
      <DeleteDialog
        open={open}
        onClose={mockOnClose}
        onConfirm={mockOnConfirm}
        title={title}
        description={description}
      />
    );
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("renders the dialog when open is true", () => {
    renderComponent(true);

    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText(description)).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /Não/i })).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /Sim/i })).toBeInTheDocument();
  });

  test("does not render the dialog when open is false", () => {
    renderComponent(false);

    expect(screen.queryByText(title)).not.toBeInTheDocument();
    expect(screen.queryByText(description)).not.toBeInTheDocument();
    expect(screen.queryByRole("button", { name: /Não/i })).not.toBeInTheDocument();
    expect(screen.queryByRole("button", { name: /Sim/i })).not.toBeInTheDocument();
  });

  test("calls onClose when 'Não' button is clicked", () => {
    renderComponent(true);

    fireEvent.click(screen.getByRole("button", { name: /Não/i }));

    expect(mockOnClose).toHaveBeenCalledTimes(1);
    expect(mockOnConfirm).not.toHaveBeenCalled();
  });

  test("calls onConfirm when 'Sim' button is clicked", () => {
    renderComponent(true);

    fireEvent.click(screen.getByRole("button", { name: /Sim/i }));

    expect(mockOnConfirm).toHaveBeenCalledTimes(1);
    expect(mockOnClose).not.toHaveBeenCalled();
  });

  test("matches snapshot when open", () => {
    const { asFragment } = renderComponent(true); // Agora funciona corretamente
    expect(asFragment()).toMatchSnapshot();
  });

  test("matches snapshot when closed", () => {
    const { asFragment } = renderComponent(false); // Agora funciona corretamente
    expect(asFragment()).toMatchSnapshot();
  });
});
