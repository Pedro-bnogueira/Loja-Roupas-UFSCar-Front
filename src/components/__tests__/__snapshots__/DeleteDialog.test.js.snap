// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteDialog Component matches snapshot when closed 1`] = `<DocumentFragment />`;

exports[`DeleteDialog Component matches snapshot when open 1`] = `<DocumentFragment />`;
